HelpSource/Classes/NotificationCenter.schelp

```diff
@@ -4,9 +4,35 @@ related:: Classes/SimpleController, Classes/NodeWatcher
 categories:: Control
 
 description::
-One common OOP pattern is Model-View-Controller where one object (the controller) is a dependant of the model. Every time the model changes it notifies all of its dependants. In this case the model has a dictionary of dependants and iterates through those.
+One common OOP pattern is the Observer Pattern where one object (the controller or observer) is a dependant of a model or subject object ( https://en.wikipedia.org/wiki/Observer_pattern ). Every time the model changes it notifies all of its dependants. In the default implementation of this pattern in SuperCollider, the model has a dictionary of dependants and iterates through those.  A limitation of this implementation is that all dependants always receive the same message "update" when the model changes.  If a dependant is required to perform different actions depending on the type of change that happens in the model, or on the dependant itself, then this must be coded in the update method of the dependant.  The NotificationCenter overcomes this limitation by making it possible to register a function to be performed for each combination of model, notification message and dependant.  There are thus 4 objects involved in each connecetion between model and dependant in the NotificationCenter:
 
-Another common pattern is NotificationCenter wherein an object emits a notification and clients can register functions that will be executed when that notification happens.
+NUMBEREDLIST::
+## The model (can be any type of object).
+## The notification message (usually a Symbol).
+## The dependant (can be any type of object).
+## The action to be performed (usually a Function).
```

Comment:
what arguments is the function called with?

Reply:
The four items provided here are the arguments. I will make that explicit in an example.

Reply:
let me say it a different way -- does the action receive any arguments when it is triggered?